File: android/app/src/main/java/com/mplauction/android/data/remote/ShortId.kt
package com.mplauction.android.data.remote

import kotlin.random.Random

private const val ID_ALPHABET = "abcdefghijklmnopqrstuvwxyz0123456789"

// A short, easy-to-read-aloud-and-type document ID — used anywhere a human
// needs to share a code (an auction ID, a draft match ID) rather than paste
// a full Firestore auto-ID.
fun generateShortId(length: Int = 6): String =
  (1..length).map { ID_ALPHABET[Random.nextInt(ID_ALPHABET.length)] }.joinToString("").uppercase()

File: src/lib/shortId.ts
// A short, easy-to-read-aloud ID for anything a human shares by typing it
// (an auction ID, a Team Draft match ID) rather than pasting a Firestore auto-ID.
export function generateShortId(): string {
  return Math.random().toString(36).slice(2, 8).toUpperCase()
}
